packages/web3-core/src/web3_config.ts

```diff
@@ -79,7 +79,7 @@ export abstract class Web3Config
 		transactionConfirmationPollingInterval: undefined,
 		blockHeaderTimeout: 10,
 		maxListenersWarningThreshold: 100,
-		contractDataInputFill: 'input',
+		contractDataInputFill: 'data',
```

Comment:
this will default to `data` for contracts package, 
did you check directly using web3-eth sendTransaction? and also implicit usage of eth_call (for getting revert reason) in sendTransaction and sendRawTransaction?

Reply:
These methods both go through the web3-utils `formatTransaction` [method](https://github.com/web3/web3.js/pull/6294/files#diff-0c52e30746e2e9181931331752f8cb162bc84dba5e7a8cd093509681c8c2707fR20) web3-eth sendTransaction and call will only use whatever the user is sending (if the user sends data, it will be formatted using data, if user uses input they will use input). 

So there is no change needed